definitions/scenarios/self_upgrade.rb

```diff
@@ -65,13 +75,26 @@ class SelfUpgrade < SelfUpgradeBase
     end
 
     def downstream_self_upgrade(pkgs_to_update)
+      unless upgrade_allowed?
+        if version_difference > 1
+          raise(
+            ForemanMaintain::Error::Warn,
+            "foreman-maintain is too many versions ahead, given the" \
+            "target version is #{target_version}. Please rollback " \
+            "foreman-maintain to the proper version."
```

Comment:
This message could use some word smithing to help get the right point across. I debated including a note about distrosync? That may only apply to EL systems?

Reply:
Maybe adding a note about what the current version is so they can see the difference? Like make it more obvious that the version is < X.Y-1